.github/workflows/release-internal.yml

```diff
@@ -31,21 +32,76 @@ on:
         type: boolean
         required: true
         default: false
+      dry_run:
+        description: "Is this a dry run? (default to false)"
+        type: boolean
+        required: true
+        default: false
+
+  schedule:
+    - cron: '0 13 * * 3' # Every Wednesday at 8:00 AM central time
 
 defaults:
   run:
     shell: bash
 
 jobs:
+  check-for-new-commits:
+    runs-on: ubuntu-latest
+    outputs:
+      latest_commit_sha: ${{ steps.check_new_commits.outputs.latest_commit_sha }}
+      has_new_commits: ${{ steps.check_new_commits.outputs.has_new_commits }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Fetch all history for all branches and tags
+
+      - name: Fetch ref
+        run: |
+          git fetch origin ${{ inputs.ref }}:${{ inputs.ref }}
+
+      - name: Get last successful Release to Cloud run details and check for new commits
+        id: check_new_commits
+        run: |
+          # Get last successful Release to Cloud run details
+          last_run_url=$(curl --silent -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/${{ github.repository }}/actions/runs?event=workflow_dispatch&status=success&branch=main&workflow_file=release-internal.yml" | jq -r '.workflow_runs[0].url')
```

Comment:
this appears to be operating unexpected, i'm targeting workflows based on the the specific file but the returned object from the dry_run was for an `adapter integration test run`

[dry run](https://github.com/dbt-labs/dbt-bigquery/actions/runs/10046774056/job/27766989444)
[URL](https://api.github.com/repos/dbt-labs/dbt-bigquery/actions/runs/10042568588)